### PR TITLE
Benchmarks: Microbenchmark - Support in-place for NCCL/RCCL benchmark

### PR DESCRIPTION
**Pull request by yzygitzh**
**Description**
Add in-place metrics for NCCL/RCCL benchmark for latency measurement.

### CODE REVIEW

File: docs/user-tutorial/benchmarks/micro-benchmarks.md [hunk 1]
```diff
@@ -292,16 +292,16 @@ Support the following traffic patterns:
...
-| nccl-bw/${operation}_${msg_size}_time  | time (us)        | NCCL operation lantency with given message size.            |
-| nccl-bw/${operation}_${msg_size}_algbw | bandwidth (GB/s) | NCCL operation algorithm bandwidth with given message size. |
-| nccl-bw/${operation}_${msg_size}_busbw | bandwidth (GB/s) | NCCL operation bus bandwidth with given message size.       |
-| rccl-bw/${operation}_${msg_size}_time  | time (us)        | RCCL operation lantency with given message size.            |
-| rccl-bw/${operation}_${msg_size}_algbw | bandwidth (GB/s) | RCCL operation algorithm bandwidth with given message size. |
-| rccl-bw/${operation}_${msg_size}_busbw | bandwidth (GB/s) | RCCL operation bus bandwidth with given message size.       |
-
-If mpi mode is enable and traffic pattern is specified, the metrics pattern will change to `nccl-bw/${operation}_${serial_index)_${parallel_index):${msg_size}_time`
+| Name                                                    | Unit             | Description                                                                            |
```
Comment (yukirora) [inline docs/user-tutorial/benchmarks/micro-benchmarks.md:295]: can we keep original metric names for out-of-place, just add new in place metrics?

Comment (cp5555) [reply docs/user-tutorial/benchmarks/micro-benchmarks.md:295]: I agree. Please don't change the original metric names.

Comment (yzygitzh) [reply docs/user-tutorial/benchmarks/micro-benchmarks.md]: Fixed.

File: superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py [hunk 2]
```diff
@@ -153,9 +153,6 @@ def _process_raw_result(self, cmd_idx, raw_output):    # noqa: C901
```
[the remark this answers is not in the document]
Comment (yzygitzh) [reply superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py]: Fixed.

File: tests/benchmarks/micro_benchmarks/test_cuda_nccl_bw_performance.py [hunk 3]
```diff
@@ -109,6 +110,9 @@ def test_nccl_bw_performance(self, allgather, allreduce, reduce, broadcast, redu
         assert (benchmark.result['alltoall_8589934592_algbw'][0] == 256.36)
         assert (benchmark.result['alltoall_8589934592_busbw'][0] == 224.31)
 
+        # Switch to in-place
```
Comment (yukirora) [inline tests/benchmarks/micro_benchmarks/test_cuda_nccl_bw_performance.py:113]: why need to change the original test, i think we'd better add a new testcase for in place?

**Review by cp5555 — APPROVED** [review]

**Review by yukirora — APPROVED** [review]

**Pull request merged**